FGEGraphics/ClientSystem/ShaderLocations.cs

```diff
@@ -42,4 +46,309 @@ public static class Common2D
+        /// Uniform: aspect
+        /// </summary>
+        public const int ASPECT_RATIO = 7;
+
+        /// <summary>
+        /// The position of the light source (sun location).
+        /// Uniform: sunloc
+        /// </summary>
+        public const int SUN_LOCATION = 8;
+
+        /// <summary>
+        /// The density of the godray effect.
+        /// Higher values make the godray effect more dense and focused while lower values make it more spread out.
+        /// Uniform: density
+        /// </summary>
+        public const int DENSITY = 12;
+
+        /// <summary>
+        /// The minimum depth used for depth linearization.
+        /// Uniform: MIN_DEPTH
+        /// </summary>
+        public const int MIN_DEPTH = 14;
+
+        /// <summary>
+        /// The maximum depth used for depth linearization.
+        /// Uniform: MAX_DEPTH
+        /// </summary>
+        public const int MAX_DEPTH = 15;
+
+        /// <summary>
+        /// Distance that determines whether a pixel is in the sky or not.
+        /// Uniform: SKY_DIST
+        /// </summary>
+        public const int SKY_DISTANCE = 16;
+    }
+
+    /// <summary>Locations used in the finalgodray shader.</summary>
+    public static class FinalGodray
+    {
+        /// <summary>
+        /// What position the camera is targeting in the world (ray traced).
+        /// Uniform: cameraTargetPos
+        /// </summary>
+        public const int CAMERA_TARGET_POSITION = 8;
+
+        /// <summary>
+        /// How far away the camera target position is from the camera.
+        /// Uniform: cameraTargetDepth
+        /// </summary>
+        public const int CAMERA_TARGET_DEPTH = 9;
+
+        /// <summary>
+        /// The HDR exposure value.
+        /// Uniform: hdrExposure
+        /// </summary>
+        public const int HDR_EXPOSURE = 10;
+
+        /// <summary>
+        /// Camera position relative to rendering origin.
+        /// Uniform: cameraPos
+        /// </summary>
+        public const int CAMERA_RELATIVE_POSITION = 14;
+
+        /// <summary>
+        /// The distance fog should be around.
+        /// Uniform: fogDist
+        /// </summary>
+        public const int FOG_DISTANCE = 16;
+
+        /// <summary>
+        /// The Z-Near and Z-Far value of the 3D projection.
+        /// Uniform: zdist
+        /// </summary>
+        public const int Z_DISTANCE = 17;
+
+        /// <summary>
+        /// What color any fog to apply is. For no fog, the alpha value will be zero.
+        /// Uniform: fogCol
+        /// </summary>
+        public const int FOG_COLOR = 18;
+
+        /// <summary>
+        /// How much to desaturation the view by. 1.0 = fully desaturated.
+        /// Uniform: desaturationAmount
+        /// </summary>
+        public const int DESATURATION = 19;
+
+        /// <summary>
+        /// What position the eye of the 3D camera view is at in the world.
```

Comment:
This is also copied from pre-existing docs but said pre-existing docs were ... not very clear. I think this is for 3DVR multi-eye rendering?